controllers/runnerdeployment_controller_test.go

```diff
@@ -189,20 +217,24 @@ var _ = Context("Inside of a new namespace", func() {
 
 				Eventually(
 					func() int {
-						err := k8sClient.List(ctx, &runnerSets, client.InNamespace(ns.Name))
+						selector, err := metav1.LabelSelectorAsSelector(rd.Spec.Selector)
 						if err != nil {
-							logf.Log.Error(err, "list runner sets")
+							logf.Log.Error(err, "failed to get RunnerDeployment selector")
+							return -1
```

Comment:
Can we instead change the signature of `func() int` to `func (int, error)` and return an error here? (I can do it after merging anyway so please not hesitate to defer it to me. Of course doing it yourself is more than welcome :)

Reply:
Confirmed. Thanks!